fvm/evm/types/events.go

```diff
@@ -21,8 +21,45 @@ const (
 )
 
 type EventPayload interface {
-	// CadenceEvent creates a Cadence event type
-	CadenceEvent() (cadence.Event, error)
+	// ToCadence converts the event to Cadence event
+	ToCadence() (cadence.Event, error)
+}
+
+type BlockEventPayload struct {
+	Height            uint64           `cadence:"height"`
+	Hash              string           `cadence:"hash"`
+	Timestamp         uint64           `cadence:"timestamp"`
+	TotalSupply       cadence.Int      `cadence:"totalSupply"`
+	ParentBlockHash   string           `cadence:"parentHash"`
+	ReceiptRoot       string           `cadence:"receiptRoot"`
+	TransactionHashes []cadence.String `cadence:"transactionHashes"`
+}
+
+// DecodeBlockEventPayload decodes Cadence event into block event payload.
+func DecodeBlockEventPayload(event cadence.Event) (*BlockEventPayload, error) {
+	var block BlockEventPayload
+	err := cadence.DecodeFields(event, &block)
+	return &block, err
+}
+
+type TransactionEventPayload struct {
+	Hash            string `cadence:"hash"`
+	Index           uint16 `cadence:"index"`
+	TransactionType uint8  `cadence:"type"`
+	Payload         string `cadence:"payload"`
+	Error           uint16 `cadence:"error"`
+	GasConsumed     uint64 `cadence:"gasConsumed"`
+	ContractAddress string `cadence:"contractAddress"`
+	Logs            string `cadence:"logs"`
+	BlockHeight     uint64 `cadence:"blockHeight"`
+	BlockHash       string `cadence:"blockHash"`
```

Comment:
If we want to further save space, I guess we can omit `BlockHash`, as it is rather lengthy. `BlockHeight` should suffice in order to correlate a `evm.TransactionExecuted` event, to its corresponding `evm.BlockExecuted` event. What do you think?

Reply:
Yeah I was thinking about that. Why I'm not super convinced is that it would create a relationship between block and tx in the evm gateway for example, tx requiring block to exist, but I guess that would be reasonable. I will add a todo, and we can revisit maybe, would like to get this next update out before this changes. For sure good idea and feedback!